cep-recipe-serialization.md

```diff
@@ -0,0 +1,1110 @@
+- Description: Contains information about subpackages that are used in
+  `pin_subpackage` expressions.
+  - `curl`: An example subpackage.
+    - `name`: The name of the subpackage (e.g., `curl`).
+    - `version`: The version of the subpackage (e.g., `8.0.1`).
+    - `build_string`: The build string for the subpackage (e.g., `h60d57d3_0`).
+
+##### `packaging_settings`
+
+- Type: object
+- Description: Contains settings related to the final package output.
+  - `archive_type`: The type of archive used for packaging (either `tar_bz2` or
+    `conda`).
+  - `compression_level`: The numerical compression level used for packaging
+    (e.g., `15`).
+  - `compression_threads`: The number of threads used for compression (e.g.,
+    `1`).
+
+#### Example
+
+```yaml
+build_configuration:
+  target_platform: osx-arm64
+  host_platform: osx-arm64
+  build_platform: osx-arm64
+  variant:
+    target_platform: osx-arm64
+  hash:
+    hash: 60d57d3
+    hash_input: '{"target_platform": "osx-arm64"}' # input values for the hash
+    hash_prefix: "" # would contain `py311` and similar prefixes based on variant config
+  directories:
+    host_prefix: # host prefix with placeholders for the host environment
+    build_prefix: # build prefix for the build environment
+    work_dir: # work dir with extracted sources
+    build_dir: # build dir that contains host_prefix, build_prefix and work_dir as subdirectories
+  channels:
+    - conda-forge
+  timestamp: 2024-04-13T14:35:30.774863Z
+  subpackages:
+    curl:
+      name: curl
+      version: 8.0.1
+      build_string: h60d57d3_0
+  packaging_settings:
+    archive_type: conda
+    compression_level: 15
+    compression_threads: 1
+```
+
+### Finalized dependencies
+
+The finalized dependencies section contains a "lockfile" of all the dependencies
+exactly as they were used at build-time, for the two environments `build` and
+`host`. It also includes the final `run` and `constrains` specs that are
+computed based on the `build` and `host` dependencies (resolved all
+`run_exports`).
+
+The `finalized_dependencies` section contains the following keys:
+
+```yaml
+finalized_dependencies:
+  build:
+    specs: [list of spec objects]
+    resolved: [list of resolved dependency objects]
+    run_exports: { package_name: run_export } # all run_exports.json that were found among the packages in this environment
+  host:
+    specs: [list of spec objects]
+    resolved: [list of resolved dependency objects]
+    run_exports: { package_name: run_export } # all run_exports.json that were found among the packages in this environment
+  run:
+    depends: [list of dependencies with dependency source information]
+    constrains: [list of constrains with dependency source information]
+    run_exports: optional run_export dictionary
+```
+
+#### Spec objects
+
+A dependency spec contains the `MatchSpec` and additional information about the
+source of the dependency. There are 6 possible `DependencyInfo` versions:
```

Comment:
This appears to be the only mention of `DependencyInfo`. It's not clear to me what its purpose is.